nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/functions/ToString.java

```
@@ -0,0 +1,95 @@
+import org.apache.nifi.serialization.record.util.DataTypeUtils;
+
+import java.nio.charset.Charset;
+import java.util.stream.Stream;
+
+public class ToString extends RecordPathSegment {
+
+    private final RecordPathSegment recordPath;
+    private final RecordPathSegment charsetSegment;
+
+    public ToString(final RecordPathSegment recordPath, final RecordPathSegment charsetSegment, final boolean absolute) {
+        super("toString", null, absolute);
+        this.recordPath = recordPath;
+        this.charsetSegment = charsetSegment;
+    }
+
+    @Override
+    public Stream<FieldValue> evaluate(RecordPathEvaluationContext context) {
+        final Stream<FieldValue> fieldValues = recordPath.evaluate(context);
+        return fieldValues.filter(fv -> fv.getValue() != null)
+                .map(fv -> {
+                    final Charset charset = getCharset(this.charsetSegment, context);
+                    Object value = fv.getValue();
+                    final String stringValue;
+
+                    if (value instanceof Object[]) {
+                        Object[] o = (Object[]) value;
+                        if (o.length > 0) {
+
+                            byte[] dest = new byte[o.length];
+                            for (int i = 0; i < o.length; i++) {
+                                dest[i] = (byte) o[i];
+                            }
+                            stringValue = new String(dest, charset);
+                        } else {
+                            stringValue = ""; // Empty array = empty string
+                        }
+                    } else if (!(fv.getValue() instanceof byte[])) {
+                        return fv;
+                    } else {
+                        try {
+                            stringValue = DataTypeUtils.toString(fv.getValue(), (String) null, charset);
+                        } catch (final Exception e) {
+                            return fv;
```

If any RuntimeException is thrown here, I don't think we want to silently ignore it. Should probably let it fly.